src/BuiltInTools/DotNetDeltaApplier/StartupHook.cs

```diff
@@ -26,9 +26,11 @@ public static void Initialize()
         // When launching the application process dotnet-watch sets Hot Reload environment variables via CLI environment directives (dotnet [env:X=Y] run).
         // Currently, the CLI parser sets the env variables to the dotnet.exe process itself, rather then to the target process.
         // This may cause the dotnet.exe process to connect to the named pipe and break it for the target process.
-        if (Path.ChangeExtension(processPath, ".exe") != Path.ChangeExtension(s_targetProcessPath, ".exe"))
+        var processExe = Path.ChangeExtension(processPath, ".exe");
```

Comment:
I don't think the extension is guaranteed on Linux.  What would you expect to happen in that case?

Reply:
If the path has no extension `.exe` is appended.

Reply:
```
> Path.ChangeExtension(@"/a/b", ".exe")
"/a/b.exe"
> Path.ChangeExtension(@"/a/b.exe", ".exe")
"/a/b.exe"
> Path.ChangeExtension(@"/a/b.dll", ".exe")
"/a/b.exe"
```